src/main/java/org/prebid/server/floors/BasicPriceFloorResolver.java

```diff
@@ -0,0 +1,457 @@
+            return null;
+        }
+
+        if (MapUtils.isEmpty(modelGroup.getValues())) {
+            return null;
+        }
+
+        final String delimiter = ObjectUtils.firstNonNull(schema.getDelimiter(), SCHEMA_DEFAULT_DELIMITER);
+        final List<String> desiredRuleKey = createRuleKey(schema, bidRequest, imp, mediaType, format);
+
+        final Map<String, BigDecimal> rules = keysToLowerCase(modelGroup.getValues());
+
+        final String rule = findRule(rules, delimiter, desiredRuleKey);
+        final BigDecimal floorForRule = rule != null ? rules.get(rule) : null;
+
+        final BigDecimal floor = floorForRule != null ? floorForRule : modelGroup.getDefaultFloor();
+
+        return resolveResult(floor, rule, floorForRule, bidRequest, modelGroup.getCurrency(), currency);
+    }
+
+    private List<String> createRuleKey(PriceFloorSchema schema,
+                                       BidRequest bidRequest,
+                                       Imp imp,
+                                       ImpMediaType mediaType,
+                                       Format format) {
+
+        return schema.getFields().stream()
+                .map(field -> toFieldValue(field, bidRequest, imp, mediaType, format))
+                .map(fieldValue -> ObjectUtils.firstNonNull(fieldValue, StringUtils.EMPTY))
+                .map(String::toLowerCase)
+                .collect(Collectors.toList());
+    }
+
+    private String toFieldValue(PriceFloorField field,
+                                BidRequest bidRequest,
+                                Imp imp,
+                                ImpMediaType mediaType,
+                                Format format) {
+        final ImpMediaType resolvedMediaType = ObjectUtils.defaultIfNull(mediaType, mediaTypeFromImp(imp));
+
+        switch (field) {
+            case siteDomain:
+                return siteDomainFromRequest(bidRequest);
+            case pubDomain:
+                return pubDomainFromRequest(bidRequest);
+            case bundle:
+                return bundleFromRequest(bidRequest);
+            case channel:
+                return channelFromRequest(bidRequest);
+            case mediaType:
+                return resolvedMediaType.toString();
+            case size:
+                return sizeFromFormat(ObjectUtils.defaultIfNull(format, formatFromImp(imp, resolvedMediaType)));
+            case gptSlot:
+                return gptAdSlotFromImp(imp);
+            case pbAdSlot:
+                return pbAdSlotFromImp(imp);
+            case country:
+                return countryFromRequest(bidRequest);
+            case devicetype:
+                return resolveDeviceTypeFromRequest(bidRequest);
+            default:
+                throw new IllegalStateException("Unknown field type");
+        }
+    }
+
+    private static String siteDomainFromRequest(BidRequest bidRequest) {
+        final Site site = bidRequest.getSite();
```

Comment:
Can be inlined. Not strict, but if agree - pls check similar cases below code.

Reply:
Don't like to many extractions in one line of code
```java
final Publisher sitePublisher = ObjectUtil.getIfNotNull(bidRequest.getSite(), Site::getPublisher);
```